### ISSUE TITLE
'name not found' using discovery service with kubernetes DNS

### ISSUE THREAD

**Issue by arbe** [hazelcast/hazelcast-kubernetes#297]
I'm having some issues setting up autodiscovery using kubernetes DNS. My setups is with an embedded cache server, a kubernetes Deployment with just two replicas, and the pods never find each other and create a single member cluster each:

```
2021-02-09 11:17:13.689  INFO 1 --- [           main] c.h.internal.cluster.ClusterService      : [192.168.42.51]:5701 [dev] [3.12.7] 
Members {size:1, ver:1} [
	Member [192.168.42.51]:5701 - 854f2fae-2450-406d-9bee-874704ae7b91 this
]
```
I configured a headless service as stated in the documentation, and configured hazelcast with service-dns property:

```
apiVersion: v1
kind: Service
metadata:
  name: myservice-headless
  labels:
    app: myservice
spec:
  selector:
    app: myservice
  type: ClusterIP
  clusterIP: None
  ports:
    - name: hazelcast
      port: 5701
```

```
            config.getNetworkConfig().getJoin().getMulticastConfig().setEnabled(false);
            config.getNetworkConfig().getJoin().getKubernetesConfig().setEnabled(true)
                .setProperty("service-dns", "myservice-headless");
```

But logs are always showing the following warning:

```
2021-02-09 11:17:12.686  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.42.51]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
```

But the name does exist indeed. I can get inside the container, and try to ping myservice-headless.develop.svc.cluster.local without any issue, even just using "myservice-headless", as the headless service is in the same namespace the pods are running...

Am I missing something on this setup?

Some more info, because I'm using hazelcast 3, I'm using hazelcast-kubernetes version 1.5.5.

**Comment by hasancelik** [comment]
Hi @arbe,

I could create Hazelcast cluster at GKE which is based on DNS discovery with below steps:

1. Create ConfigMap via using below file, my namespace is default so if you use different namespace, do not forget to update `service-dns`:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: hazelcast-dns-configuration
  labels:
    app.kubernetes.io/instance: hz-dns
    app.kubernetes.io/name: hazelcast
data:
  hazelcast.yaml: |-
    hazelcast:
      network:
        join:
          kubernetes:
            enabled: true
            service-dns: hz-dns-hazelcast.default.svc.cluster.local
          multicast:
            enabled: false
        rest-api:
          enabled: true
          endpoint-groups:
            HEALTH_CHECK:
              enabled: true
      management-center:
        enabled: false
```
2. Deploy cluster via using helm:
```
$ helm install hz-dns --set hazelcast.existingConfigMap=hazelcast-dns-configuration \ 
--set mancenter.enabled=false --set image.tag=3.12.7  \
--version 2.10.3 hazelcast/hazelcast
```


```
$ kgpo

NAME                 READY   STATUS    RESTARTS   AGE
hz-dns-hazelcast-0   1/1     Running   0          3m23s
hz-dns-hazelcast-1   1/1     Running   0          2m43s
hz-dns-hazelcast-2   1/1     Running   0 
```
```
$ klo hz-dns-hazelcast-0

INFO: [10.164.1.14]:5701 [dev] [3.12.7] Kubernetes Discovery activated with mode: DNS_LOOKUP
Feb 09, 2021 2:57:28 PM com.hazelcast.instance.Node
INFO: [10.164.1.14]:5701 [dev] [3.12.7] Activating Discovery SPI Joiner
Feb 09, 2021 2:57:28 PM com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl
INFO: [10.164.1.14]:5701 [dev] [3.12.7] Starting 2 partition threads and 3 generic threads (1 dedicated for priority tasks)
Feb 09, 2021 2:57:28 PM com.hazelcast.internal.diagnostics.Diagnostics
INFO: [10.164.1.14]:5701 [dev] [3.12.7] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
Feb 09, 2021 2:57:28 PM com.hazelcast.core.LifecycleService
INFO: [10.164.1.14]:5701 [dev] [3.12.7] [10.164.1.14]:5701 is STARTING
Feb 09, 2021 2:57:28 PM com.hazelcast.spi.discovery.integration.DiscoveryService
WARNING: [10.164.1.14]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'hz-dns-hazelcast.default.svc.cluster.local' failed: name not found
Feb 09, 2021 2:57:28 PM com.hazelcast.spi.discovery.integration.DiscoveryService
WARNING: [10.164.1.14]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'hz-dns-hazelcast.default.svc.cluster.local' failed: name not found
Feb 09, 2021 2:57:28 PM com.hazelcast.spi.discovery.integration.DiscoveryService
WARNING: [10.164.1.14]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'hz-dns-hazelcast.default.svc.cluster.local' failed: name not found
Feb 09, 2021 2:57:28 PM com.hazelcast.spi.discovery.integration.DiscoveryService
WARNING: [10.164.1.14]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'hz-dns-hazelcast.default.svc.cluster.local' failed: name not found
Feb 09, 2021 2:57:29 PM com.hazelcast.spi.discovery.integration.DiscoveryService
WARNING: [10.164.1.14]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'hz-dns-hazelcast.default.svc.cluster.local' failed: name not found
Feb 09, 2021 2:57:29 PM com.hazelcast.spi.discovery.integration.DiscoveryService
WARNING: [10.164.1.14]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'hz-dns-hazelcast.default.svc.cluster.local' failed: name not found
...

Members {size:3, ver:3} [
        Member [10.164.1.14]:5701 - 04668b6b-e0dc-4622-88b8-50f092e73bc2 this
        Member [10.164.2.5]:5701 - 8a5a6f5c-9178-462d-b46d-ba0312e2a15c
        Member [10.164.1.15]:5701 - ccfa5c79-2e0f-4d7e-ae5d-c07fbf6daebb
]
```

As already mentioned at https://github.com/hazelcast/hazelcast-kubernetes/issues/185, warning logs above does not exist at second pod(hz-dns-hazelcast-1)'s logs.

Could you compare above setup with yours or directly try them at your env then inform me about the result?

**Comment by arbe** [comment]
Hi @hasancelik, and thanks for getting back on this,

The setup above is quite different indeed, as you are deploying directly hazelcast in client-server topology I believe, whereas I am deploying hazelcast embedded in my springboot application.

Second difference is that I'm not using hazelcast.xml to configure these embedded hazelcast instances, but an equivalent Java-config setup.

```
            config.getNetworkConfig().getJoin().getMulticastConfig().setEnabled(false);
            config.getNetworkConfig().getJoin().getKubernetesConfig().setEnabled(true)
                .setProperty("service-dns", "myservice-headless.develop.svc.cluster.local");
```
In my case, these warning logs appear in both pods. I'm adding below the complete logs from one of the pods

```
2021-02-10 08:50:04.796 DEBUG 1 --- [           main] c.s.a.config.CacheConfiguration   : Configuring Hazelcast
2021-02-10 08:50:04.968  INFO 1 --- [           main] com.hazelcast.instance.AddressPicker     : [LOCAL] [dev] [3.12.7] Prefer IPv4 stack is true, prefer IPv6 addresses is false
2021-02-10 08:50:04.982  INFO 1 --- [           main] com.hazelcast.instance.AddressPicker     : [LOCAL] [dev] [3.12.7] Picked [192.168.53.93]:5701, using socket ServerSocket[addr=/0.0.0.0,localport=5701], bind any local is true
2021-02-10 08:50:04.996  INFO 1 --- [           main] com.hazelcast.system                     : [192.168.53.93]:5701 [dev] [3.12.7] Hazelcast 3.12.7 (20200430 - bf4b643) starting at [192.168.53.93]:5701
2021-02-10 08:50:04.996  INFO 1 --- [           main] com.hazelcast.system                     : [192.168.53.93]:5701 [dev] [3.12.7] Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
2021-02-10 08:50:05.704  INFO 1 --- [           main] c.h.s.i.o.impl.BackpressureRegulator     : [192.168.53.93]:5701 [dev] [3.12.7] Backpressure is disabled
2021-02-10 08:50:06.703  INFO 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] Kubernetes Discovery properties: { service-dns: myservice-headless.develop.svc.cluster.local, service-dns-timeout: 5, service-name: null, service-port: 0, service-label: null, service-label-value: true, namespace: default, pod-label: null, pod-label-value: null, resolve-not-ready-addresses: false, use-node-name-as-external-address: false, kubernetes-api-retries: 3, kubernetes-master: https://kubernetes.default.svc}
2021-02-10 08:50:06.713  INFO 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] Kubernetes Discovery activated with mode: DNS_LOOKUP
2021-02-10 08:50:06.811  INFO 1 --- [           main] com.hazelcast.instance.Node              : [192.168.53.93]:5701 [dev] [3.12.7] Activating Discovery SPI Joiner
2021-02-10 08:50:07.113  INFO 1 --- [           main] c.h.s.i.o.impl.OperationExecutorImpl     : [192.168.53.93]:5701 [dev] [3.12.7] Starting 2 partition threads and 3 generic threads (1 dedicated for priority tasks)
2021-02-10 08:50:07.168  INFO 1 --- [           main] c.h.internal.diagnostics.Diagnostics     : [192.168.53.93]:5701 [dev] [3.12.7] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
2021-02-10 08:50:07.182  INFO 1 --- [           main] com.hazelcast.core.LifecycleService      : [192.168.53.93]:5701 [dev] [3.12.7] [192.168.53.93]:5701 is STARTING
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.hazelcast.internal.networking.nio.SelectorOptimizer (file:/app/libs/hazelcast-3.12.7.jar) to field sun.nio.ch.SelectorImpl.selectedKeys
WARNING: Please consider reporting this to the maintainers of com.hazelcast.internal.networking.nio.SelectorOptimizer
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
2021-02-10 08:50:07.309  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:07.332  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:07.368  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:07.410  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:07.493  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:07.659  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:07.981  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:08.482  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:08.985  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:09.487  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:09.991  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:10.492  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:10.995  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:11.496  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:11.997  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.53.93]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'myservice-headless.develop.svc.cluster.local' failed: name not found
2021-02-10 08:50:12.499  INFO 1 --- [           main] c.h.internal.cluster.ClusterService      : [192.168.53.93]:5701 [dev] [3.12.7] 

Members {size:1, ver:1} [
	Member [192.168.53.93]:5701 - 0f5203d5-49c8-45c7-a42e-ca2ba5aa64c5 this
]
```

And the warning messages keep poping up in both pods forever every two minutes, for example on the other pod:

```
2021-02-10 08:59:39.107  WARN 1 --- [cached.thread-2] c.h.s.d.integration.DiscoveryService     : [192.168.36.205]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'simplrapi-headless.develop.svc.cluster.local' failed: name not found
```

**Comment by hasancelik** [comment]
@arbe,

I have tried [embedded topology with spring-boot](https://guides.hazelcast.org/kubernetes-embedded/) at GKE but I could not reproduce your issue:

```
$ kg all

NAME                          READY   STATUS    RESTARTS   AGE
pod/my-app-784db5c9c6-hcd6h   1/1     Running   0          8m22s
pod/my-app-784db5c9c6-l5n6n   1/1     Running   0          8m22s

NAME                       TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
service/hz-dns-hazelcast   ClusterIP   None         <none>        5701/TCP   23m
service/kubernetes         ClusterIP   10.168.0.1   <none>        443/TCP    29m

NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/my-app   2/2     2            2           17m

NAME                                DESIRED   CURRENT   READY   AGE
replicaset.apps/my-app-784db5c9c6   2         2         2       17m
``` 


```
$ klo my-app-784db5c9c6-hcd6h

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                (v2.4.1)

2021-02-12 15:04:07.998  INFO 1 --- [           main] g.h.springboot.HazelcastApplication      : Starting HazelcastApplication v0.1 using Java 1.8.0_212 on my-app-784db5c9c6-hcd6h with PID 1 (/app.jar startedby root in /)
2021-02-12 15:04:08.011  INFO 1 --- [           main] g.h.springboot.HazelcastApplication      : No active profile set, falling back to default profiles: default
2021-02-12 15:04:10.337  INFO 1 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2021-02-12 15:04:10.372  INFO 1 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2021-02-12 15:04:10.373  INFO 1 --- [           main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.41]
2021-02-12 15:04:10.491  INFO 1 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
....
2021-02-12 15:04:14.104  INFO 1 --- [cached.thread-3] com.hazelcast.nio.tcp.TcpIpConnector     : [10.164.0.6]:5701 [dev] [3.12.7] Connecting to /10.164.1.5:5701, timeout: 10000, bind-any: true
2021-02-12 15:04:14.119  INFO 1 --- [.IO.thread-in-0] com.hazelcast.nio.tcp.TcpIpConnection    : [10.164.0.6]:5701 [dev] [3.12.7] Initialized new cluster connection between /10.164.0.6:5701 and /10.164.1.5:56017
2021-02-12 15:04:14.156  INFO 1 --- [.IO.thread-in-1] com.hazelcast.nio.tcp.TcpIpConnection    : [10.164.0.6]:5701 [dev] [3.12.7] Initialized new cluster connection between /10.164.0.6:47339 and /10.164.1.5:5701
2021-02-12 15:04:19.191  INFO 1 --- [           main] c.h.internal.cluster.ClusterService      : [10.164.0.6]:5701 [dev] [3.12.7]

Members {size:1, ver:1} [
        Member [10.164.0.6]:5701 - 2d29b458-ef65-4ae1-a116-c5137c21c4dc this
]

2021-02-12 15:04:19.242  INFO 1 --- [           main] com.hazelcast.core.LifecycleService      : [10.164.0.6]:5701 [dev] [3.12.7] [10.164.0.6]:5701 is STARTED
2021-02-12 15:04:19.573  INFO 1 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2021-02-12 15:04:20.013  INFO 1 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8080 (http) with context path ''
2021-02-12 15:04:20.053  INFO 1 --- [           main] g.h.springboot.HazelcastApplication      : Started HazelcastApplication in 12.849 seconds (JVM running for 13.789)
2021-02-12 15:04:25.493  INFO 1 --- [ration.thread-0] c.h.internal.cluster.ClusterService      : [10.164.0.6]:5701 [dev] [3.12.7]

Members {size:2, ver:2} [
        Member [10.164.0.6]:5701 - 2d29b458-ef65-4ae1-a116-c5137c21c4dc this
        Member [10.164.1.5]:5701 - 3e52b99a-bceb-4585-ab56-17df98d1f14a
]
```

I think the issue is related to your service's selector(`app: myservice`). You need to configure your app/deployment name as a selector, not your service name:
```
$ kd service/hz-dns-hazelcast

Name:              hz-dns-hazelcast
Namespace:         default
Labels:            <none>
Annotations:       cloud.google.com/neg: {"ingress":true}
Selector:          app=my-app  <------- check here ------>
Type:              ClusterIP
IP Families:       <none>
IP:                None
IPs:               <none>
Port:              hazelcast  5701/TCP
TargetPort:        5701/TCP
Endpoints:         10.164.0.6:5701,10.164.1.5:5701
Session Affinity:  None
Events:            <none>
```

**Comment by arbe** [comment]
Thanks, I'll have a look and double check the configuration on my deployment/service selector. Although, if there were some issue there, DNS wouldn't be resolved neither from the container interactive shell right? In addition, I already have a service with the very same selector for my app (for port 8080 in that case, not headless, which I use to expose my app through an Ingress).

Also, are you sure you are using DNS lookup in this example you tried? 
If I open the link above, it seems a page leading to use "Kubernetes API" rather than "DNS lookup" automatic discovery setup.

**Comment by hasancelik** [comment]
> Although, if there were some issue there, DNS wouldn't be resolved neither from the container interactive shell right?

I removed `selector` from my service and try to resolve its DNS at hazelcast container, it did not work:

```
/ # nslookup hz-dns-hazelcast.default.svc.cluster.local
nslookup: can't resolve '(null)': Name does not resolve
nslookup: can't resolve 'hz-dns-hazelcast.default.svc.cluster.local': Name does not resolve
```
Then I added selector into the service, redeploy it and try again:
```
/ # nslookup hz-dns-hazelcast.default.svc.cluster.local
Name:      hz-dns-hazelcast.default.svc.cluster.local
Address 1: 10.164.0.7 my-app-784db5c9c6-ld27m
Address 2: 10.164.1.6
```

> Also, are you sure you are using DNS lookup in this example you tried?

I have used DNS lookup based yaml file for the application not example yaml, here it is:
```
apiVersion: v1
kind: Service
metadata:
  name: hz-dns-hazelcast
spec:
  type: ClusterIP
  clusterIP: None
  selector:
    app: my-app
  ports:
  - name: hazelcast
    port: 5701
```

**Comment by arbe** [comment]
Hi @hasancelik, and thanks again for all the support on this!

TLDR; everything is working as expected, not sure what I had wrong exactly.

So after reviewing the configuration on my service, and seeing everything ok, I just decided to increase again the replicas to two (I had one because of this issue indeed), and then I've seen this on the already running pod:

```
2021-02-15 15:50:50.139  INFO 1 --- [.IO.thread-in-0] com.hazelcast.nio.tcp.TcpIpConnection    : [192.168.43.55]:5701 [dev] [3.12.7] Initialized new cluster connection between /192.168.43.55:5701 and /192.168.31.182:53203
2021-02-15 15:50:57.141  INFO 1 --- [ration.thread-0] c.h.internal.cluster.ClusterService      : [192.168.43.55]:5701 [dev] [3.12.7] 

Members {size:2, ver:2} [
	Member [192.168.43.55]:5701 - 9c3519da-3ccb-4dde-9106-1b31bcd01bee this
	Member [192.168.31.182]:5701 - 79c6f86d-aaba-4fa2-be14-d96fc4046260
]


2021-02-15 15:50:57.416  INFO 1 --- [r-api.migration] c.h.i.partition.impl.MigrationManager    : [192.168.43.55]:5701 [dev] [3.12.7] Re-partitioning cluster data... Migration queue size: 271
```

And of course the new pod already starting with this 2 member cluster. So I wonder what is it I had wrong before? I did a quick test of restarting both pods at the same time, and by doing that I could get back the WARN messages in both pods on startup:

```
2021-02-15 15:58:16.799  WARN 1 --- [           main] c.h.s.d.integration.DiscoveryService     : [192.168.23.214]:5701 [dev] [3.12.7] DNS lookup for serviceDns 'simplrapi-headless.develop.svc.cluster.local' failed: name not found
```

And they started their own cache of 1 member, and it took a bit more than 5 minutes to have each pod seeing each other through the DNS, so they could finally create the cluster of two members.

So I still do not know what happened before, if there was something wrong with Kubernetes DNS or what, as I have literally changed nothing in the configuration (other than reducing/increasing replicas), and I was more patient than just waiting 5 minutes to see everything working.

Anyhow, happy to see this working now. Feel free to close it...

**Comment by hasancelik** [comment]
@arbe Glad to hear that your issue solved mysteriously 🙂 

If you encounter this issue again, feel free to re-open it 👍

**Comment by Anton-Trixter** [comment]
Greetings to you all.

I think, I have the same problem when starting hazelcast cluster.
Here is the error message that I have:
`
WARNING: [10.236.40.224]:5701 [dev] [5.3.6] DNS lookup for serviceDns 'datahub-gms-v0-13-0-hazelcast-svc' failed: unknown host
`

The hazelcast cluster starts as a part of Datahub GMS service.

This is the yaml of Hazelcast service:
```
apiVersion: v1
kind: Service
metadata:
  name: {{ printf "%s-%s-%s" .Release.Name (regexReplaceAll "[^-a-z0-9]+" .Values.imageTag "-") "hazelcast-svc" | trunc 63 | trimSuffix "-" }}
spec:
  clusterIP: None
  ports:
    - name: hazelcast
      port: 5701
      protocol: TCP
      targetPort: 5701
  selector:
    app.kubernetes.io/name: {{- include "datahub-gms.name" . | nindent 6 }}
  type: ClusterIP
```

I tried to start the cluster consisting of 2 Pods, but no luck - the Pods hangs for some time (status 0/1) and after that they restarts.
If I ping Pods from one to another everything is ok.

Any suggestions to fix this?
